src/basics.md

```diff
@@ -326,13 +327,77 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-run-piped-external-commands]: #ex-run-piped-external-commands
+<a name="ex-run-piped-external-commands"></a>
+## Run piped external commands
+
+[![std-badge]][std] [![cat-os-badge]][cat-os]
+
+Shows up to the 10<sup>th</sup> biggest files and subdirectories in
+the specified the current working directory. It spawns three Unix
+processes as external [`Command`]. It is equivalant to run: `du -ah . | sort -hr | head -n
+10`.
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+#
+use std::process::{Command, Stdio};
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#         Utf8(std::string::FromUtf8Error);
+#     }
+# }
+#
+fn run() -> Result<()> {
+    let directory = ".";
+    let du_process = Command::new("du")
```

Comment:
We might be tempted to further DRY the code by extracting the builder into a separate fn like 

```rust
fn spawn_piped<P, I, A, IN>(program: P, args: I, input: IN) -> Result<Child>
where
    P: AsRef<OsStr>,
    I: IntoIterator<Item = A>,
    A: AsRef<OsStr>,
    IN: Into<Stdio>,
{
...
}
```

But I'm not sure if this does not make the code harder to read. I'll let you decide here

Reply:
I gave it a try:
* [`pipe-process.rs`](https://gist.github.com/ludwigpacifici/d94dc241c2362270e9950aced26545dd)
* [`pipe-process-factorized.rs`](https://gist.github.com/ludwigpacifici/bd67799a15e8624008f7bcf268d482cc)

I prefer `pipe-process.rs`:
1. The goal is to demonstrate pipe processes, however, the `spawn_piped` function tends to bury it.
2. If the targeted reader is a beginner, `spawn_piped` introduce concepts of iterators. 
3. `spawn_piped` will DRY the code if more processes would be spawn.

If I miss something, let me know, otherwise I will skip `spawn_piped`.